core/state/statedb.go

```diff
@@ -575,17 +574,25 @@ func (s *StateDB) DeleteUTXO(txHash common.Hash, outputIndex uint16) {
 }
 
 // CreateUTXO explicitly creates a UTXO entry.
-func (s *StateDB) CreateUTXO(txHash common.Hash, outputIndex uint16, utxo *types.UtxoEntry) {
+func (s *StateDB) CreateUTXO(txHash common.Hash, outputIndex uint16, utxo *types.UtxoEntry) error {
 	if metrics_config.MetricsEnabled() {
 		defer func(start time.Time) { stateMetrics.WithLabelValues("CreateUTXO").Add(float64(time.Since(start))) }(time.Now())
 	}
+	// This check is largely redundant, but it's a good sanity check. Might be removed in the future.
+	if err := common.CheckIfBytesAreInternalAndQiAddress(utxo.Address, s.nodeLocation); err != nil {
+		return err
+	}
+	if utxo.Denomination > types.MaxDenomination { // sanity check
```

Comment:
if you are doing this, might as well add a check for less than MinDenomination

Reply:
MinDenomination is zero

Reply:
A uint can't be zero, so the check is implicit

Reply:
fair point